DirectEmployers/MyJobs
# [PD-2207] Rework Company Faceting

## Overview

This PR is pretty simple. It changes out the logic for when a company_slug is used to narrow a set of job results. Originally, the logic would involve using company names stored in BUIDs. The new logic uses the slug to match against the beginning of the company_slab_exact solr field.

## Testing

The best way to verify that this code is working properly the following:

* Load a large number of jobs into your local DB
* Navigate to www.my.jobs.
* On the right bar, note the companies and their job counts.
* Click various companies and ensure that the number displayed on the filter label is the same as the number of jobs returned.
